packages/core/src/browser/saveable.ts

```diff
@@ -85,6 +85,7 @@ export namespace Saveable {
                         throw e;
                     }
                 } finally {
+                    widget.close = close;
```

Comment:
could you elaborate on this change?

Reply:
If the widget's constructor is not called when the widget is reopened, this will prevent adding new overrided  close handler each time when the widget is reopened. Removed that because don't need this fix after overriding dispose() in the preferences widgets.